### PR TITLE
Jian ssh new

### PR DESCRIPTION
**Pull request by xujianin1988**
Add ssh support for Redshell and ssh unit test.

### CODE REVIEW

**Comment by xujianin1988** [comment]
Note: this version of ssh server did not support tab completion. We need to create another issue for that.

File: ssh-util.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+import base64
+import pty
+import multiprocessing
+import argparse
+import signal
+
+log.startLogging(sys.stderr)
+
+# Path to RSA SSH keys used by the server.
+SERVER_RSA_PRIVATE = 'ssh_conf/ssh_host_rsa_key'
```
Comment (zimventures) [inline ssh-util.py]: Can you please open an issue to make these variables user configurable via the settings file? Most applications will want to store their certificates outside of the CPDK directory.

File: ssh-util.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        self.terminal.write("Starting process '" + self.process_obj.process_cmd + "'...")
+        self.terminal.nextLine()
+        self.process_obj.process_start()
+        output_thread = Thread(target=self.output_redirect)
+        output_thread.start()
+        time.sleep(.3)
+
+    def connectionLost(self, reason):
+
+        print "Reason: " + str(reason)
```
Comment (zimventures) [inline ssh-util.py]: remove, or change to a logging call

File: ssh-util.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+            self.process = pexpect.spawn(self.process_cmd)
+            self.process.setecho(False)
+            self.read_flg = True
+            time.sleep(.1)
+            thread = Thread(target=self.read_thread)
+            thread.start()
+            time.sleep(.3)
+
+        def read_thread(self):
```
Comment (zimventures) [inline ssh-util.py]: Does this loop at 100%? I'm worried this will beat up the CPU.

File: ssh-util.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+                    return True
+                else:
+                    return False
+            else:
+                return False
+
+        def force_stop(self):
+
+            if self.process.isalive() is True:
+                print "Force stop process starts..."
```
Comment (zimventures) [inline ssh-util.py]: Change to a logging call or remove.

File: ssh-util.py (hunk 3 above)
Comment (xujianin1988) [inline ssh-util.py]: No the while loop has two checks. One is read_flg and the other one is process.isalive(). Once the process is shut down or forced quit, this loop will end, also I used the non-blocking read, so it won't stuck there.

File: ssh-util.py (hunk 1 above)
Comment (xujianin1988) [inline ssh-util.py]: Okay I will open an issue for that

**Pull request merged**